### ISSUE TITLE
YNAB Importer: expects parameter 1 to be string, null given in GenerateTransactions.php

### ISSUE THREAD

**Issue by blinkiz** [firefly-iii/firefly-iii#3694]
expects parameter 1 to be string, null given in GenerateTransactions.php
Using 5 days old docker image fireflyiii/ynab-importer:develop. Meaning v2.0.0 but develop branch.

Let me know if you need any more information. :slightly_smiling_face: 
```
[2020-08-13 13:46:20] local.DEBUG: Going to try and access https://firefly.example.com  
[2020-08-13 13:46:26] local.DEBUG: Now at App\Http\Controllers\Import\UploadController::upload  
[2020-08-13 13:46:26] local.DEBUG: Now at App\Http\Controllers\Import\BudgetController::index  
[2020-08-13 13:46:31] local.DEBUG: Now at App\Http\Controllers\Import\BudgetController::postIndex  
[2020-08-13 13:46:31] local.DEBUG: Now at App\Http\Controllers\Import\ConfigurationController::index  
[2020-08-13 13:46:59] local.DEBUG: Now at App\Http\Controllers\Import\ConfigurationController::postIndex  
[2020-08-13 13:46:59] local.DEBUG: Constructed ImportRoutineManager  
[2020-08-13 13:46:59] local.DEBUG: Was given no download identifier, will generate one.  
[2020-08-13 13:46:59] local.DEBUG: Going to generate download identifier.  
[2020-08-13 13:46:59] local.DEBUG: Attempt #1 results in "8Ea5VnQV9n23o23e"  
[2020-08-13 13:46:59] local.INFO: Download job identifier is "8Ea5VnQV9n23o23e"  
[2020-08-13 13:46:59] local.DEBUG: Download job file does not exist or error, create a new one.  
[2020-08-13 13:46:59] local.DEBUG: Return download job status. {"status":"waiting_to_start","errors":[],"warnings":[],"messages":[]} 
[2020-08-13 13:46:59] local.DEBUG: Download routine manager identifier is "8Ea5VnQV9n23o23e"  
[2020-08-13 13:46:59] local.DEBUG: Stored "8Ea5VnQV9n23o23e" under "download_job_id"  
[2020-08-13 13:47:00] local.DEBUG: Now at App\Http\Controllers\Import\DownloadController::start  
[2020-08-13 13:47:00] local.DEBUG: Constructed ImportRoutineManager  
[2020-08-13 13:47:00] local.DEBUG: Was given download identifier, will use it.  
[2020-08-13 13:47:00] local.DEBUG: Now in download setJobStatus(job_running) for job 8Ea5VnQV9n23o23e  
[2020-08-13 13:47:00] local.DEBUG: Now in storeJobStatus(8Ea5VnQV9n23o23e): job_running  
[2020-08-13 13:47:00] local.DEBUG: Download ImportRoutineManager: created XXXXXX with download identifier "8Ea5VnQV9n23o23e"  
[2020-08-13 13:47:00] local.DEBUG: Now in App\Ynab\Download\RoutineManager::start  
[2020-08-13 13:47:00] local.DEBUG: Now in getTransaction()  
[2020-08-13 13:47:00] local.DEBUG: Now downloading from budget d8b8849a-xxxx-xxxx-xxxx-99f27bb49a0f  
[2020-08-13 13:47:00] local.DEBUG: Going to download from account 284af611-xxxx-xxxx-8079-08b7c70d195c  
[2020-08-13 13:47:01] local.DEBUG: Found 139 transaction(s) in account 284af611-xxxx-xxxx-8079-08b7c70d195c  
[2020-08-13 13:47:01] local.DEBUG: Going to download from account bd01aac9-xxxx-xxxx-9541-5bab8492d490  
[2020-08-13 13:47:01] local.DEBUG: Found 61 transaction(s) in account bd01aac9-xxxx-xxxx-9541-5bab8492d490  
[2020-08-13 13:47:01] local.DEBUG: Merged into 200 total transactions.  
[2020-08-13 13:47:01] local.DEBUG: Now in download setJobStatus(job_done) for job 8Ea5VnQV9n23o23e  
[2020-08-13 13:47:01] local.DEBUG: Now in storeJobStatus(8Ea5VnQV9n23o23e): job_done  
[2020-08-13 13:47:01] local.DEBUG: SyncController is creating new routine manager with NEW sync identifier  
[2020-08-13 13:47:01] local.DEBUG: Constructed RoutineManager for sync  
[2020-08-13 13:47:01] local.DEBUG: Going to generate sync job identifier.  
[2020-08-13 13:47:01] local.DEBUG: Attempt #1 results in "EmuaFd6HByisQJcw"  
[2020-08-13 13:47:01] local.INFO: Sync job identifier is "EmuaFd6HByisQJcw"  
[2020-08-13 13:47:01] local.DEBUG: Sync file does not exist or error, create a new one.  
[2020-08-13 13:47:01] local.DEBUG: Sync routine manager job identifier is "EmuaFd6HByisQJcw"  
[2020-08-13 13:47:01] local.DEBUG: Stored "EmuaFd6HByisQJcw" under "sync_job_id"  
[2020-08-13 13:47:09] local.DEBUG: Constructed RoutineManager for sync  
[2020-08-13 13:47:09] local.DEBUG: Now in Sync setJobStatus(job_running)  
[2020-08-13 13:47:09] local.DEBUG: Found "EmuaFd6HByisQJcw" in the session  
[2020-08-13 13:47:09] local.DEBUG: Now in Sync storeJobStatus(EmuaFd6HByisQJcw): job_running  
[2020-08-13 13:47:09] local.DEBUG: Done with storing.  
[2020-08-13 13:47:09] local.DEBUG: Now in App\Services\Sync\RoutineManager::start  
[2020-08-13 13:47:09] local.DEBUG: Going to parse YNAB download.  
[2020-08-13 13:47:09] local.INFO: Message: 8Ea5VnQV9n23o23e 0: Decoded YNAB download.  
[2020-08-13 13:47:09] local.DEBUG: Now in Sync storeJobStatus(8Ea5VnQV9n23o23e): job_done  
[2020-08-13 13:47:09] local.DEBUG: Done with storing.  
[2020-08-13 13:47:09] local.DEBUG: Parsed 600 YNAB account transactions.  
[2020-08-13 13:47:09] local.DEBUG: Done parsing YNAB download.  
[2020-08-13 13:47:09] local.ERROR: strpos() expects parameter 1 to be string, null given {"exception":"[object] (TypeError(code: 0): strpos() expects parameter 1 to be string, null given at /var/www/html/app/Services/Sync/GenerateTransactions.php:241)
[stacktrace]
...
```

**Comment by JC5** [comment]
Thanks, that's a bug. I'll pick it up and release a new version!

**Comment by JC5** [comment]
Fixed in 2.0.1.

**Comment by blinkiz** [comment]
Nicely done. Impressive.
I have tested and the error is now gone. Thanks!
